### PR TITLE
Add tags property to agent type

### PR DESCRIPTION
**Pull request by gabrielbussolo**
# Summary of this proposal

Add tag field on aggregator, agent and pipelines typess

## Issue(s) number(s)

fix #66 
## Checklist for submitter

- [ ] My PR has a related issue/bug number.
- [ ] My PR provides tests
  - [ ] Integration tests are added/passes
  - [ ] Unit tests are added/passes
  - [ ] End-to-end tests added/passes
  - [ ] Static code check added/passes
  - [ ] Linting on documentation added/passes
  - [ ] Does not affect code coverage stats
- [ ] My PR requires updating dependencies
- [ ] My PR has the documentation changes required.

## Checklist for reviewer

- [ ] The proposal fixes a bug/issue or implements a new feature that is well described.
- [ ] The proposal has sufficient test cases that covers the changes.
  - [ ] If changes an API, it does not break backwards compatibility (-1 major version).
- [ ] If integration is required, the proposal has integration tests.
- [ ] The proposal does not break coverage stats
- [ ] The proposal has the required documentation changes.

## Backport

- [ ] Backport to the latest stable release.

### CODE REVIEW

File: types/agent.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type Agent struct {
 	MetricsCount        uint64           `json:"metricsCount" yaml:"metricsCount"`
 	CreatedAt           time.Time        `json:"createdAt" yaml:"createdAt"`
 	UpdatedAt           time.Time        `json:"updatedAt" yaml:"updatedAt"`
+	Tags                []string         `json:"tags" yaml:"tags"`
```
Comment (nicolasparada) [inline types/agent.go:26]: Can you move this up? Before the timestamps.

We are somehow ordering them by ID, required fields, optional fields, timestamps, optional relationships.

**Review by nicolasparada — COMMENTED** [review]
If you could order them and move tags before timestamp and stuff.
Also I noticed there are no option to **update** tags.
That's ok tho. We can update tags for later.

File: types/agent.go [hunk 2]
```diff
@@ -76,6 +78,7 @@ type AgentsParams struct {
 	Last   *uint64
 	Before *string
 	Name   *string
+	Tags   *[]string
```
Comment (nicolasparada) [inline types/agent.go:81]: Oh, also what does it mean to pass a list of many tags?
Is it an `AND` or an `OR`?

I mean, a user could ask for all agents that have the tags `foo` **and** `bar`.
Or a user could ask for all the agents that either have the tag `foo` **or** `bar`.

Or a user could ask (front-end) for a complex query like `(tag=foo && tag=bar) || tag=baz`. How will we able to provide an API that supports that? 🤔 

Maybe we should allow a single `tag` string to do the filtering.
Then in a second future iteration we could add some kind of simple expresion parsing to this tag string to support that kind of list matching.

Comment (nicolasparada) [reply types/agent.go:81]: CC @niedbalski

Comment (niedbalski) [reply types/agent.go:81]: This was intended to be the first implementation for AND listing, I agree with @nicolasparada the type should be a string not and array as it doesn't allows future NOT, negation or any other sub-expressions, that string will be parseable by the API, for now  only AND operator and can be separated either by commas or AND  (I would prefer AND) https://github.com/calyptia/cloud/issues/401#issuecomment-1075018109

Comment (nicolasparada) [reply types/agent.go:81]: I don't know if supporting `AND` yet. What about quotes and stuff.
There are edge cases we haven't talked over yet.
I suggest just filtering by one single tag for now.

Comment (niedbalski) [reply types/agent.go:81]: > 

something fancier to trim to [a-z]+ and split by AND?

Comment (gabrielbussolo) [reply types/agent.go:81]: sorry, guys i couldn't visualize the concern. i understood that the structure itself is an array and to send it through the client i created this other two functions associated with the type (on another branch for now):

```go
func (a *AgentsParams) TagsAsString() string {
	return strings.Join(*a.Tags, ",")
}

func (a *AgentsParams) TagFilter(tags ...string) {
	a.Tags = &tags
}
```
thinking to accomplish:
```go
opt.TagFilter("value1", "value2")
```


and using it to set on url params like

```go
if params.Tags != nil {
q.Set("tags", params.TagsAsString())
}
```

i couldnt understand how if its "or", "and" affects on client level, i thought that its a problem on repository level

**Pull request merged**